### PR TITLE
[DOCS] Fixing Painless tests

### LINKED ISSUE #68101: [CI] DocsClientYamlTestSuiteIT.test {yaml=reference/rest-api/usage/line_39} failure
**Build scan**:

https://gradle-enterprise.elastic.co/s/gkwrfu7vcedre

**Repro line**:

```
./gradlew ':docs:integTest' --tests "org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT.test {yaml=reference/rest-api/usage/line_39}" \
  -Dtests.seed=F198448862F3D5DE \
  -Dtests.security.manager=true \
  -Dbuild.snapshot=false \
  -Dtests.jvm.argline="-Dbuild.snapshot=false" \
  -Dtests.locale=cs-CZ \
  -Dtests.timezone=America/Ensenada \
  -Druntime.java=11
```

**Reproduces locally?**:

Yes

**Applicable branches**:

master, 7.x, 7.11

**Failure history**:

Five failures in the last 12h

**Failure excerpt:**

```
java.lang.RuntimeException: Failure at [reference/rest-api/usage:16]: stashed value not found for key [body.watcher.execution.actions.logging.total]
	at __randomizedtesting.SeedInfo.seed([F198448862F3D5DE:79CC7B52CC0FB826]:0)
	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:418)
	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:393)
```
```
[2021-01-28T00:15:13,093][INFO ][o.e.s.DocsClientYamlTestSuiteIT] [test] [yaml=reference/rest-api/usage/line_39] before test
[2021-01-28T00:15:13,263][INFO ][o.e.s.DocsClientYamlTestSuiteIT] [test] Stash dump on test failure [{
  "stash" : {
    "body" : {
      "aggregate_metric" : {
        "available" : true,
        "enabled" : true
      },
      "analytics" : {
        "available" : true,
        "enabled" : true,
        "stats" : {
          "boxplot_usage" : 0,
          "cumulative_cardinality_usage" : 0,
          "string_stats_usage" : 2,
          "top_metrics_usage" : 0,
          "t_test_usage" : 1,
          "moving_percentiles_usage" : 0,
          "normalize_usage" : 0,
          "rate_usage" : 1
        }
      },
      "ccr" : {
        "available" : true,
        "enabled" : true,
        "follower_indices_count" : 0,
        "auto_follow_patterns_count" : 0
      },
      "data_streams" : {
        "available" : true,
        "enabled" : true,
        "data_streams" : 0,
        "indices_count" : 0
      },
      "data_tiers" : {
        "available" : true,
        "enabled" : true,
        "data_warm" : {
          "node_count" : 1,
          "index_count" : 0,
          "total_shard_count" : 0,
          "primary_shard_count" : 0,
          "doc_count" : 0,
          "total_size_bytes" : 0,
          "primary_size_bytes" : 0,
          "primary_shard_size_avg_bytes" : 0,
          "primary_shard_size_median_bytes" : 0,
          "primary_shard_size_mad_bytes" : 0
        },
        "data_cold" : {
          "node_count" : 1,
          "index_count" : 0,
          "total_shard_count" : 0,
          "primary_shard_count" : 0,
          "doc_count" : 0,
          "total_size_bytes" : 0,
          "primary_size_bytes" : 0,
          "primary_shard_size_avg_bytes" : 0,
          "primary_shard_size_median_bytes" : 0,
          "primary_shard_size_mad_bytes" : 0
        },
        "data_content" : {
          "node_count" : 1,
          "index_count" : 0,
          "total_shard_count" : 0,
          "primary_shard_count" : 0,
          "doc_count" : 0,
          "total_size_bytes" : 0,
          "primary_size_bytes" : 0,
          "primary_shard_size_avg_bytes" : 0,
          "primary_shard_size_median_bytes" : 0,
          "primary_shard_size_mad_bytes" : 0
        },
        "data_hot" : {
          "node_count" : 1,
          "index_count" : 0,
          "total_shard_count" : 0,
          "primary_shard_count" : 0,
          "doc_count" : 0,
          "total_size_bytes" : 0,
          "primary_size_bytes" : 0,
          "primary_shard_size_avg_bytes" : 0,
          "primary_shard_size_median_bytes" : 0,
          "primary_shard_size_mad_bytes" : 0
        }
      },
      "eql" : {
        "available" : true,
        "enabled" : true,
        "features" : {
          "joins" : {
            "join_queries_three" : 0,
            "join_queries_two" : 0,
            "join_until" : 0,
            "join_queries_five_or_more" : 0,
            "join_queries_four" : 0
          },
          "sequence" : 0,
          "keys" : {
            "join_keys_two" : 0,
            "join_keys_one" : 0,
            "join_keys_three" : 0,
            "join_keys_five_or_more" : 0,
            "join_keys_four" : 0
          },
          "join" : 0,
          "sequences" : {
            "sequence_queries_three" : 0,
            "sequence_queries_four" : 0,
            "sequence_queries_two" : 0,
            "sequence_until" : 0,
            "sequence_queries_five_or_more" : 0,
            "sequence_maxspan" : 0
          },
          "event" : 0,
          "pipes" : {
            "pipe_tail" : 0,
            "pipe_head" : 0
          }
        },
        "queries" : {
          "all" : {
            "total" : 0,
            "failed" : 0
          },
          "_all" : {
            "total" : 0,
            "failed" : 0
          }
        }
      },
      "frozen_indices" : {
        "available" : true,
        "enabled" : true,
        "indices_count" : 0
      },
      "graph" : {
        "available" : true,
        "enabled" : true
      },
      "ilm" : {
        "policy_count" : 7,
        "policy_stats" : [
          {
            "phases" : {
              "hot" : {
                "min_age" : 0,
                "actions" : [
                  "rollover"
                ]
              }
            },
            "indices_managed" : 0
          },
          {
            "phases" : {
              "delete" : {
                "min_age" : 604800000,
                "actions" : [
                  "delete"
                ]
              }
            },
            "indices_managed" : 0
          },
          {
            "phases" : {
              "hot" : {
                "min_age" : 0,
                "actions" : [
                  "rollover"
                ]
              }
            },
            "indices_managed" : 0
          },
          {
            "phases" : {
              "hot" : {
                "min_age" : 0,
                "actions" : [
                  "rollover"
                ]
              }
            },
            "indices_managed" : 0
          },
          {
            "phases" : {
              "hot" : {
                "min_age" : 0,
                "actions" : [
                  "rollover"
                ]
              }
            },
            "indices_managed" : 0
          },
          {
            "phases" : {
              "hot" : {
                "min_age" : 0,
                "actions" : [
                  "rollover"
                ]
              }
            },
            "indices_managed" : 0
          },
          {
            "phases" : {
              "hot" : {
                "min_age" : 0,
                "actions" : [
                  "rollover"
                ]
              },
              "delete" : {
                "min_age" : 7776000000,
                "actions" : [
                  "delete"
                ]
              }
            },
            "indices_managed" : 0
          }
        ]
      },
      "logstash" : {
        "available" : true,
        "enabled" : true
      },
      "ml" : {
        "available" : true,
        "enabled" : true,
        "jobs" : {
          "_all" : {
            "count" : 0,
            "detectors" : {
              "total" : 0.0,
              "min" : 0.0,
              "avg" : 0.0,
              "max" : 0.0
            },
            "created_by" : { },
            "model_size" : {
              "total" : 0.0,
              "min" : 0.0,
              "avg" : 0.0,
              "max" : 0.0
            },
            "forecasts" : {
              "total" : 0,
              "forecasted_jobs" : 0
            }
          }
        },
        "datafeeds" : {
          "_all" : {
            "count" : 0
          }
        },
        "data_frame_analytics_jobs" : {
          "_all" : {
            "count" : 0
          }
        },
        "inference" : {
          "ingest_processors" : {
            "_all" : {
              "num_docs_processed" : {
                "max" : 0,
                "sum" : 0,
                "min" : 0
              },
              "pipelines" : {
                "count" : 0
              },
              "num_failures" : {
                "max" : 0,
                "sum" : 0,
                "min" : 0
              },
              "time_ms" : {
                "max" : 0,
                "sum" : 0,
                "min" : 0
              }
            }
          },
          "trained_models" : {
            "_all" : {
              "count" : 0
            }
          }
        },
        "node_count" : 1
      },
      "monitoring" : {
        "available" : true,
        "enabled" : true,
        "collection_enabled" : false,
        "enabled_exporters" : {
          "local" : 1
        }
      },
      "rollup" : {
        "available" : true,
        "enabled" : true
      },
      "runtime_fields" : {
        "available" : true,
        "enabled" : true,
        "field_types" : [ ]
      },
      "searchable_snapshots" : {
        "available" : true,
        "enabled" : true,
        "indices_count" : 0
      },
      "security" : {
        "available" : true,
        "enabled" : false,
        "ssl" : {
          "http" : {
            "enabled" : false
          },
          "transport" : {
            "enabled" : false
          }
        }
      },
      "slm" : {
        "available" : true,
        "enabled" : true
      },
      "spatial" : {
        "available" : true,
        "enabled" : true
      },
      "sql" : {
        "available" : true,
        "enabled" : true,
        "features" : {
          "having" : 0,
          "subselect" : 0,
          "limit" : 0,
          "orderby" : 1,
          "where" : 0,
          "groupby" : 0,
          "join" : 0,
          "local" : 0,
          "command" : 0
        },
        "queries" : {
          "rest" : {
            "total" : 2,
            "paging" : 1,
            "failed" : 0
          },
          "cli" : {
            "total" : 0,
            "paging" : 0,
            "failed" : 0
          },
          "canvas" : {
            "total" : 0,
            "paging" : 0,
            "failed" : 0
          },
          "odbc" : {
            "total" : 0,
            "paging" : 0,
            "failed" : 0
          },
          "jdbc" : {
            "total" : 0,
            "paging" : 0,
            "failed" : 0
          },
          "odbc32" : {
            "total" : 0,
            "paging" : 0,
            "failed" : 0
          },
          "odbc64" : {
            "total" : 0,
            "paging" : 0,
            "failed" : 0
          },
          "_all" : {
            "total" : 2,
            "paging" : 1,
            "failed" : 0
          },
          "translate" : {
            "count" : 0
          }
        }
      },
      "transform" : {
        "available" : true,
        "enabled" : true
      },
      "vectors" : {
        "available" : true,
        "enabled" : true,
        "dense_vector_fields_count" : 0,
        "dense_vector_dims_avg_count" : 0
      },
      "voting_only" : {
        "available" : true,
        "enabled" : true
      },
      "watcher" : {
        "available" : true,
        "enabled" : true,
        "execution" : {
          "actions" : {
            "_all" : {
              "total" : 0,
              "total_time_in_ms" : 0
            }
          }
        },
        "watch" : {
          "input" : {
            "_all" : {
              "total" : 0,
              "active" : 0
            }
          },
          "trigger" : {
            "_all" : {
              "total" : 0,
              "active" : 0
            }
          }
        },
        "count" : {
          "total" : 0,
          "active" : 0
        }
      }
    }
  }
}]
[2021-01-28T00:15:13,314][INFO ][o.e.s.DocsClientYamlTestSuiteIT] [test] [yaml=reference/rest-api/usage/line_39] after test
```

### PR DESCRIPTION
**Pull request by lockewritesdocs**
This PR introduces fixes for #68101:
* Moves shared examples to the [watcher transform page](https://elasticsearch_68157.docs-preview.app.elstc.co/guide/en/elasticsearch/painless/master/painless-watcher-transform-context.html) so that they aren't tested twice
* Adds a document to the integrated test so that more watcher actions are satisfied
* Fixes several watcher tests to produce the intended output

Closes #68101

### CODE REVIEW

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

**Comment by stu-elastic** [comment]
@elasticmachine run elasticsearch-ci/1
@elasticmachine run elasticsearch-ci/2
@elasticmachine run elasticsearch-ci/bwc
@elasticmachine run elasticsearch-ci/default-distro
@elasticmachine run elasticsearch-ci/docs
@elasticmachine run elasticsearch-ci/eql-correctness
@elasticmachine run elasticsearch-ci/oss-distro-docs
@elasticmachine run elasticsearch-ci/packaging-sample-unix
@elasticmachine run elasticsearch-ci/packaging-sample-windows

**Comment by stu-elastic** [comment]
@elasticmachine run 1
@elasticmachine run 2
@elasticmachine run bwc
@elasticmachine run default-distro
@elasticmachine run docs
@elasticmachine run eql-correctness
@elasticmachine run oss-distro-docs
@elasticmachine run packaging-sample-unix
@elasticmachine run packaging-sample-windows

**Comment by stu-elastic** [comment]
@elasticmachine run bwc

**Comment by lockewritesdocs** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by lockewritesdocs** [comment]
@elasticmachine run 1
@elasticmachine run 2
@elasticmachine run bwc
@elasticmachine run default-distro
@elasticmachine run docs
@elasticmachine run eql-correctness
@elasticmachine run oss-distro-docs
@elasticmachine run packaging-sample-unix
@elasticmachine run packaging-sample-windows

File: docs/reference/rest-api/usage.asciidoc [hunk 1]
```diff
@@ -360,7 +360,6 @@ GET /_xpack/usage
   }
 }
 ------------------------------------------------------------
-// TESTRESPONSE[skip:Awaits fix https://github.com/elastic/elasticsearch/issues/68101]
```
Comment (stu-elastic) [inline docs/reference/rest-api/usage.asciidoc:363]: This is still failing.

Comment (lockewritesdocs) [reply docs/reference/rest-api/usage.asciidoc:363]: I ran the following test locally, and it succeeded:
```
./gradlew ':docs:integTest' --tests "org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT.test"
```

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by stu-elastic** [comment]
To check: is watcher running async with schedule 24hr in execute?

**Comment by jrodewig** [comment]
> To check: is watcher running async with schedule 24hr in execute?

I think @stu-elastic may be right. There may be something odd about the watcher timing. However, I think there's a workaround using [response filtering](https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#common-options-response-filtering). This method succeeded for me locally using both `./gradlew -pdocs integTest --tests "*rest-api/usage*"` and `./gradlew ':docs:integTest' --tests "org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT.test"`

1. Update the `_xpack/usage` API snippet to filter out `watcher.execution.actions.logging*`. If the fields are in the response, this will remove them. If the fields aren't in the response, it won't raise an error.

```
[source,console]
----
GET /_xpack/usage
----
// TEST[s/usage/usage?filter_path=-watcher.execution.actions.logging*/]
```

2. Remove the `watcher.execution.actions.logging*` bit from the API response. I only included the top bit here:

```
[source,console-result]
----
{
  "security" : {
    "available" : true,
    "enabled" : false,
    "ssl" : {
      "http" : {
        "enabled" : false
      },
      "transport" : {
        "enabled" : false
      }
    }
  },
  "monitoring" : {
    "available" : true,
    "enabled" : true,
    "collection_enabled" : false,
    "enabled_exporters" : {
      "local" : 1
    }
  },
  "watcher" : {
    "available" : true,
    "enabled" : true,
    "execution" : {
      "actions" : {
        "_all" : {
          "total" : 0,
          "total_time_in_ms" : 0
        }
      }
    },
    "watch" : {
...
```

**Comment by lockewritesdocs** [comment]
Thanks for the help on this one @jrodewig! I made those changes and local tests are passing. I'll push the changes and let the CI builds run again.

**Comment by lockewritesdocs** [comment]
@stu-elastic, all CI builds passed with the latest changes. Do you think this PR can be merged now? 🤔

**Review by stu-elastic — APPROVED** [review]
This fixes the error referenced in https://github.com/elastic/elasticsearch/issues/68101, looks great @lockewritesdocs.

**Pull request merged**